src/fsharp/LexFilter.fs

```diff
@@ -422,9 +415,7 @@ type TokenTup =
     member x.EndPos = x.LexbufState.EndPos
     
     /// Returns a token 'tok' with the same position as this token
-    member x.UseLocation(tok) = 
-        let tokState = x.LexbufState 
-        TokenTup(tok,LexbufState(tokState.StartPos, tokState.EndPos,false),x.LastTokenPos)
+    member x.UseLocation(tok) = TokenTup(tok,x.LexbufState,x.LastTokenPos)
```

Comment:
When looking at this, it felt like a bug, why would the EOF state change at the same location for a copy of token, I switched this in and everything passed on tests so felt was ok to integrate as saved the allocation, Most of the time it will not be EOF (false) so I can do a further commit where it checks if EOF, and if not, it will reuse state, otherwise, create a new one with the false flag?

Reply:
Yea, most of the time it will be EOF.

I think we should put in the check just in case. Also, put a comment describing why we are doing it.